hmsc/updaters/updateZ.py

```diff
@@ -104,29 +104,32 @@ def updateZ(params, data, rLHyperparams, *,
 
     ZStack = tf.concat([ZNormal, ZProbit, ZPoisson], -1)
     iDStack = tf.concat([iDNormal, iDProbit, iDPoisson], -1)
-    indColStack = tf.concat([indColNormal, indColProbit, indColPoisson], 0)
-    ZNew = tf.gather(ZStack, tf.argsort(indColStack), axis=-1)
-    iDNew = tf.gather(iDStack, tf.argsort(indColStack), axis=-1)
+    indColStack = np.concatenate([indColNormal, indColProbit, indColPoisson], 0)
+    indices = np.argsort(indColStack)
+    ZNew = tf.gather(ZStack, indices, axis=-1)
+    iDNew = tf.gather(iDStack, indices, axis=-1)
     return ZNew, iDNew, poisson_omega
 
 
 def gather(*args, indices):
     return (tf.gather(a, indices, axis=-1) for a in args)
 
 
+@tf_named_func("z_normal")
 def calculate_z_normal(Y, Yo, L, sigma, *, dtype):
     # no data augmentation for normal model in columns with continious unbounded data
     Z = tf.where(Yo, Y, L + sigma * tfr.normal(Y.shape, dtype=dtype))
     iD = tf.cast(Yo, dtype) * sigma**-2
     return Z, iD
 
 
+@tf_named_func("z_probit")
 def calculate_z_probit(Y, Yo, L, sigma, *, truncated_normal_library, dtype):
     # Albert and Chib (1993) data augemntation for probit model in columns with binary data
-    INFTY = 1e+3
+    INFTY = tf.constant(1e+3, dtype)
     Ym = tfm.logical_not(Yo)
-    low = tf.where(tfm.logical_or(Y == 0, Ym), tf.cast(-INFTY, dtype), tf.zeros_like(Y))
-    high = tf.where(tfm.logical_or(Y == 1, Ym), tf.cast(INFTY, dtype), tf.zeros_like(Y))
+    low = tf.where(tfm.logical_or(Y == 0, Ym), -INFTY, 0.)
```

Comment:
I suggest to set dtype of 0. explicitly to be on the safe side.

Reply:
Done 3d9829457db04c140422fc2be6696d6f5b0adbde.